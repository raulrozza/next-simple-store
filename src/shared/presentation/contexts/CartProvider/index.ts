export { default as useAddToCart } from './hooks/useAddToCart';
