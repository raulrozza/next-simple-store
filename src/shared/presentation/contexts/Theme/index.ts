export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
