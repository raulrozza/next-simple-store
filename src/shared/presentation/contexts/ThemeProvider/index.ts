export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
