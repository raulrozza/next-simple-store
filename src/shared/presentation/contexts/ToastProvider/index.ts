export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
