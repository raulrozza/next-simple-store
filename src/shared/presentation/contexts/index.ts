export * from './CartProvider';
export * from './ThemeProvider';
