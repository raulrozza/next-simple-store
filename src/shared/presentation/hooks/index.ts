import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
