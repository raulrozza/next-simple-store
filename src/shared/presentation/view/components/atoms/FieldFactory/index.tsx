import { FC, memo } from 'react';

import { IField } from '@/shared/presentation/entities/Field';
import Input from '@/shared/presentation/view/components/atoms/Input';
import Select from '@/shared/presentation/view/components/atoms/Select';
import TextArea from '@/shared/presentation/view/components/atoms/TextArea';

const Component = {
    Input: Input as FC<Omit<IField, 'type'>>,
    TextArea: TextArea as FC<Omit<IField, 'type'>>,
    Select: Select as FC<Omit<IField, 'type'>>,
};

const FieldFactory: FC<IField> = ({ type, ...props }) => {
    if (type === 'text') return <Component.Input {...props} />;
    if (type === 'textarea') return <Component.TextArea {...props} />;
    if (type === 'select') return <Component.Select {...props} />;

    throw new Error(`Field type ${type} is not supported`);
};

export default memo(FieldFactory);
