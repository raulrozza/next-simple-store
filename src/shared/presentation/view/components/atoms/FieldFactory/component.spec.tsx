import { render, screen } from '@testing-library/react';
import { Formik } from 'formik';

import { ThemeProvider } from '@/shared/presentation/contexts';

import FieldFactory from './index';

describe('FieldFactory', () => {
    it('should render the components correctly with all the possible types', () => {
        const view = render(
            <Formik initialValues={{}} onSubmit={jest.fn()}>
                <>
                    <FieldFactory
                        type="textarea"
                        placeholder="Large text"
                        name="textarea"
                    />
                    <FieldFactory type="text" name="text" placeholder="Text" />
                    <FieldFactory
                        type="select"
                        name="select"
                        placeholder="Select"
                        options={[]}
                    />
                </>
            </Formik>,
            { wrapper: ThemeProvider },
        );

        expect(screen.getByPlaceholderText('Large text')).toBeInTheDocument();
        expect(screen.getByPlaceholderText('Text')).toBeInTheDocument();
        expect(screen.getByPlaceholderText('Select')).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });

    it('should throw when trying to instance an invalid field type', async () => {
        expect(() =>
            render(
                <Formik initialValues={{}} onSubmit={jest.fn()}>
                    <FieldFactory
                        type={'invalid-type' as any}
                        {...({} as any)}
                    />
                </Formik>,
                { wrapper: ThemeProvider },
            ),
        ).toThrow('Field type invalid-type is not supported');
    });
});
