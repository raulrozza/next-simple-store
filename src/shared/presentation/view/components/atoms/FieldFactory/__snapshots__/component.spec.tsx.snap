// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldFactory should render the components correctly with all the possible types 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
}

.c1:focus {
  border-color: #8CCFDE;
}

.c2 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
}

.c2:focus {
  border-color: #8CCFDE;
}

.c0 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  height: 6rem;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
  resize: none;
}

.c0:focus {
  border-color: #8CCFDE;
}

<body>
    <div>
      <div>
        <textarea
          class="c0"
          name="textarea"
          placeholder="Large text"
        />
      </div>
      <div>
        <input
          class="c1"
          name="text"
          placeholder="Text"
          value=""
        />
      </div>
      <div>
        <select
          class="c2"
          name="select"
        >
          <option
            value=""
          >
            Select
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": .c1 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
}

.c1:focus {
  border-color: #8CCFDE;
}

.c2 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
}

.c2:focus {
  border-color: #8CCFDE;
}

.c0 {
  background-color: #FFFFFF;
  border: 2px solid #E6EAEA;
  outline: none;
  border-radius: 0.25rem;
  width: 100%;
  height: 6rem;
  padding: 0.5rem;
  -webkit-transition: outline 0.2s,border 0.2s;
  transition: outline 0.2s,border 0.2s;
  resize: none;
}

.c0:focus {
  border-color: #8CCFDE;
}

<div>
    <div>
      <textarea
        class="c0"
        name="textarea"
        placeholder="Large text"
      />
    </div>
    <div>
      <input
        class="c1"
        name="text"
        placeholder="Text"
        value=""
      />
    </div>
    <div>
      <select
        class="c2"
        name="select"
      >
        <option
          value=""
        >
          Select
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
